charts/k8s-monitoring/charts/feature-pod-logs/templates/_filelog.alloy.tpl

```diff
@@ -0,0 +1,158 @@
+{{- define "feature.podLogs.filelog.alloy" }}
+otelcol.receiver.filelog "pod_logs" {
+{{- if .Values.namespaces }}
+  include = [
+{{- range .Values.namespaces }}
+    "/var/log/pods/{{ . }}_*/*/*.log",
+{{- end }}
+  ]
+{{- else }}
+  include = ["/var/log/pods/*/*/*.log"]
+{{- end }}
+{{- if .Values.excludeNamespaces }}
+  exclude = [
+{{- range .Values.excludeNamespaces }}
+    "/var/log/pods/{{ . }}_*/*/*.log",
+{{- end }}
+  ]
+{{- end }}
+  start_at = {{ if .Values.filelogGatherSettings.onlyGatherNewLogLines }}"end"{{ else }}"beginning"{{ end }}
+  include_file_name = false
+  include_file_path = true
+
+  operators = [
+    // Container operator will set k8s.pod.name, k8s.pod.uid, k8s.container.name, k8s.container.restart_count, and k8s.namespace.name
+    {
+      type                       = "container",
+      add_metadata_from_filepath = true,
+    },
+  ]
+
+  output {
+    logs = [otelcol.processor.k8sattributes.pod_logs.input]
+  }
+}
+
+otelcol.processor.k8sattributes "pod_logs" {
+  pod_association {
+    source {
+      from = "resource_attribute"
+      name = "k8s.pod.uid"
+    }
+  }
+
+  extract {
+    metadata = [
+      "k8s.deployment.name",
+      "k8s.statefulset.name",
+      "k8s.daemonset.name",
+      "k8s.cronjob.name",
+      "k8s.job.name",
+      "k8s.node.name",
+    ]
+    annotation {
+      key_regex = "(.*)"
+      tag_name  = "$1"
+      from      = "pod"
+    }
+    annotation {
+      key_regex = "resource.opentelemetry.io/(.*)"
+      tag_name  = "$1"
+      from      = "pod"
+    }
```

Comment:
it seems like these could potentially conflict. I'd be interested to understand the behavior if there's a `service.name` annotation set, and a `resource.opentelemetry.io/service.name` annotation.